### PR TITLE
Navigation stack

### PR DESCRIPTION
**Pull request by ivanpauno**
Added all navigation related packages.
Also added some needed libraries for them:
- Bullet
- SDL
- SDL_image

It's all building without problems for me, please @jubeira check.

### CODE REVIEW

File: patches/bullet.patch [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+ 
+ ENDIF (WIN32)
+ 
+-
+-OPTION(BUILD_CPU_DEMOS "Build original Bullet CPU examples"        ON)
+-
++IF (NOT ANDROID)
++	OPTION(BUILD_CPU_DEMOS "Build original Bullet CPU examples"        ON)
++ELSE ()
++        OPTION(BUILD_CPU_DEMOS "Don't build original Bullet CPU examples"  OFF)
```
Comment (meyerj) [inline patches/bullet.patch]: nit: The doc string should be the same in both cases, even if the default value is `OFF` on Android.

File: patches/map_server.patch [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
++    find_package(yaml-cpp REQUIRED)
++    add_definitions(-DHAVE_YAMLCPP_GT_0_5_0)
++else()
++  pkg_check_modules(YAML_CPP yaml-cpp REQUIRED)
++  if(YAMLCPP_VERSION VERSION_GREATER "0.5.0")
+     add_definitions(-DHAVE_YAMLCPP_GT_0_5_0)
++  endif()
++  link_directories(${YAML_CPP_LIBRARY_DIRS})
+ endif()
++include_directories(${YAML_CPP_INCLUDE_DIRS})
```
Comment (meyerj) [inline patches/map_server.patch]: Also fix/patch `${YAMLCPP_LIBRARIES}` further down, which will be unset with this patch, no?

**Comment by jubeira** [comment]
I'm having an issue when building `map_server`:
```
Errors     << map_server:make /opt/ros_android/output/catkin_ws/logs/map_server/build.make.000.log                                                                                                         
CMakeFiles/map_server.dir/src/main.cpp.o: In function `MapServer':
/opt/ros_android/output/catkin_ws/src/navigation/map_server/src/main.cpp:84: undefined reference to `YAML::Load(std::__ndk1::basic_istream<char, std::__ndk1::char_traits<char> >&)'
```

I believe the problem is related to the patch applied to `map_server`'s CMakeLists (see https://github.com/Intermodalics/ros_android/pull/39#discussion_r238336967?)

File: patches/map_server.patch (hunk 2 above)
Comment (ivanpauno) [reply patches/map_server.patch]: Yes, right. I corrected it, and also I corrected transitive link dependencies in https://github.com/Intermodalics/ros_android/pull/39/commits/1eafe21388a4ab93cbf7d2332642edd23a32ac2d

**Comment by ivanpauno** [comment]
> I'm having an issue when building `map_server`:
> 
> ```
> Errors     << map_server:make /opt/ros_android/output/catkin_ws/logs/map_server/build.make.000.log                                                                                                         
> CMakeFiles/map_server.dir/src/main.cpp.o: In function `MapServer':
> /opt/ros_android/output/catkin_ws/src/navigation/map_server/src/main.cpp:84: undefined reference to `YAML::Load(std::__ndk1::basic_istream<char, std::__ndk1::char_traits<char> >&)'
> ```
> I believe the problem is related to the patch applied to `map_server`'s CMakeLists (see [#39 (comment)](https://github.com/Intermodalics/ros_android/pull/39#discussion_r238336967)?)

Solved with https://github.com/Intermodalics/ros_android/commit/1eafe21388a4ab93cbf7d2332642edd23a32ac2d

**Comment by jubeira** [comment]
Works now, thanks! Please solve the conflicts and this comment here https://github.com/Intermodalics/ros_android/pull/39/files#r238335830; besides that I think it's ready to be merged.

**Pull request merged**